spec/requests/register_api/applications_controller_spec.rb

```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
+
+RSpec.describe 'GET /register-api/applications', type: :request, sidekiq: true do
+  it 'verifies the API token' do
+    get '/register-api/applications', headers: {}
+
+    expect(response).to have_http_status(:unauthorized)
+  end
+
+  it 'does not allow access to the API from other data users' do
+    api_token = APIUser.test_data_user.create_magic_link_token!
+
+    headers = { 'Authorization' => "Bearer #{api_token}" }
```

Comment:
:+1: We can eventually fit this out with an `APISpecHelper` along the lines of https://github.com/DFE-Digital/apply-for-teacher-training/commit/8bbcc11ca74f4f022b2025667aa64453c8fc9656